geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+
+Validation of this list of listeners MUST behave the same as if the list were part of a single "parent" Gateway.
+
+eg.
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  infrastructure:
+    allowedChildren:
+      namespaces:
+        from: Same
+  gatewayClassName: example
+  listeners:
+  - name: HTTP
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: child-gateway
+spec:
+  gatewayClassName: example
+  infrastructure:
+    attachTo:
+      name: parent-gateway
+      kind: Gateway
+      group: gateway.networking.k8s.io
+  listeners:
+  - name: metrics
+    port: 8080
+    protocol: HTTP
+```
+
+With this configuration, the realized "parent" Gateway should listen on port `80` and `8080`.
+
+###### Listener Precedence
+
+Gateway Listeners should be merged using the following precedence:
+- "parent" Gateway
+- "child" Gateway ordered by creation time (oldest first)
+- "child" Gateway ordered alphabetical by “{namespace}/{name}”.
+
+If there are conflicts between these, this should be reported as `Conflicted=True` in the listener as usual.
+
+### Status Fields
+
+#### Addresses
+
+The list of `Addresses` that appear in the status of the "child" Gateway MUST be the same as the "parent" Gateway.
+
+#### AttachedGateways
+
+The property `attachedGateways` will appear on the status of the "parent" Gateway will indicate the total number of "child" Gateways that have are attached
+
+#### Gateway Conditions
+
+Gateway conditions currently supports the following condition types: `Accepted` and `Programmed`
+
+For parent gateways, `Accepted` should be set based on the entire set of merged listeners.
+For instance, if a child listener is invalid, `ListenersNotValid` would be reported.
+`Programmed` is not expected, generally, to depend on the children resources, but if an implementation does depend on these
+they should consider child resources when reporting this status.
+
+For child gateways, `Accepted` and `Programmed` should not consider the overall merged Gateway status, but only the child's own listeners.
+The exception to this is if the parent Gateway is not valid in any regard.
+
+For example, if I have a `parent`, `child-1`, and `child-2`:
+* If `parent` is entirely invalid (for example, an invalid `address`), all three Gateways will reported `Accepted=False`.
+* If `child-1` has an invalid listener, `parent` and `child-1` will report `ListenersNotValid`, while `child-2` will not.
+* If `child-1` references a parent that doesn't exist then `child-1` will report `Accepted=False`
+* If `child-1` references a parent that doesn't allow merging then `child-1` will report `Accepted=False`
+* If `child-1` references another child (eg. `child-2`) then `child-1` will report `Accepted=False` 
```

Comment:
This should be disallowed to begin with, unless a parent Gateway can transform into a child gateway.

Reply:
It is - hence why we're saying the child gateway shouldn't be accepted.

> unless a parent Gateway can transform into a child gateway.

It cannot.